### PR TITLE
Worldscreen key bindings

### PR DESCRIPTION
**Pull request by SomeTroglodyte**
These try to be faithful to default Civ5 key bindings as found in [several places](https://duckduckgo.com/?q=%22Civilization+5%22+hotkeys+%22keyboard+shortcuts%22) online (but not the fandom wiki) and I coded "something" for most I found documented.

Still, this is meant as **basis for discussion**. Every opinion welcome, e.g. on:
* Which of these would we want which not
* How to get closer to best practice / elegant code (framework first?)
* How to inform players

Obvious problems:
* Code repetition - I took the Zoom keys as example that the actual execution should be unified with touch-ui sources of the same 'user command'
* (Quick)Save/Load implementation
    * is at least the third independent implementation of almost identical routines
    * doesn't belong here
    * but - hey - it works :grinning:
* No player-accessible documentation
    * The overview button decoration is just an example (separate commit)
    * Similar hints on the other buttons more or less directly corresponding to a key would range from bah (Diplomacy) to atrocious (tech) in aesthetics
    * I'd actually suggest no such decoration and setting up a civilopedia entry instead
* Didn't fix the typo in ZoomableScrollPane continousScrollingX

### CODE REVIEW

File: core/src/com/unciv/ui/worldscreen/WorldScreen.kt [hunk 1]
```diff
@@ -175,8 +181,68 @@ class WorldScreen(val gameInfo: GameInfo, val viewingCiv:CivilizationInfo) : Cam
...
+            val toast = ToastPopup("Quicksaving...", this)
+            thread(name = "SaveGame") {
+                GameSaver.saveGame(gameInfo, "QuickSave") {
+                    Gdx.app.postRunnable {
+                        toast.close()
+                        if (it != null)
+                            ToastPopup("Quicksave failed!", this)
+                        else {
+                            ToastPopup("Quicksave OK.", this)
+                            UncivGame.Current.setWorldScreen()
```
Comment (yairm210) [inline core/src/com/unciv/ui/worldscreen/WorldScreen.kt:194]: Why is this necessary? We're already on the world screen, and it won't change because of a quicksave.

File: core/src/com/unciv/ui/worldscreen/WorldScreen.kt [hunk 2]
```diff
@@ -175,8 +181,68 @@ class WorldScreen(val gameInfo: GameInfo, val viewingCiv:CivilizationInfo) : Cam
...
+        val quickSave = {
+            val toast = ToastPopup("Quicksaving...", this)
+            thread(name = "SaveGame") {
+                GameSaver.saveGame(gameInfo, "QuickSave") {
+                    Gdx.app.postRunnable {
+                        toast.close()
+                        if (it != null)
+                            ToastPopup("Quicksave failed!", this)
+                        else {
+                            ToastPopup("Quicksave OK.", this)
```
Comment (yairm210) [inline core/src/com/unciv/ui/worldscreen/WorldScreen.kt:193]: We already have "failed to save game" and "Game saved" strings, try to reuse them to minimize translation additions

File: core/src/com/unciv/ui/worldscreen/WorldScreen.kt [hunk 3]
```diff
@@ -175,8 +181,68 @@ class WorldScreen(val gameInfo: GameInfo, val viewingCiv:CivilizationInfo) : Cam
...
+                            UncivGame.Current.setWorldScreen()
+                        }
+                    }
+                }
+            }
+            Unit
```
Comment (yairm210) [inline core/src/com/unciv/ui/worldscreen/WorldScreen.kt:199]: ???

File: core/src/com/unciv/ui/worldscreen/WorldScreenTopBar.kt [hunk 4]
```diff
@@ -143,9 +145,12 @@ class WorldScreenTopBar(val worldScreen: WorldScreen) : Table() {
...
     }
 
-    private fun getOverviewButton(): TextButton {
-        val overviewButton = "Overview".toTextButton()
-        overviewButton.labelCell.pad(10f)
+    private fun getOverviewButton(): Button {
+        val overviewButton = Button(CameraStageBaseScreen.skin)
+        overviewButton.add("Overview".toLabel()).pad(10f)
+        if (Gdx.app.input.isPeripheralAvailable(Input.Peripheral.HardwareKeyboard)) {
+            overviewButton.add(Label("(E)", CameraStageBaseScreen.skin).apply { color = Color.LIGHT_GRAY }).padLeft(0f)
```
Comment (yairm210) [inline core/src/com/unciv/ui/worldscreen/WorldScreenTopBar.kt:152]: ("E").toLabel(color = Color.LIGHT_GRAY)

**Comment by yairm210** [comment]
We *definitely* need a 'shortcuts' page to detail these keybindings...

File: core/src/com/unciv/ui/worldscreen/WorldScreen.kt (hunk 1 above)
Comment (SomeTroglodyte) [reply core/src/com/unciv/ui/worldscreen/WorldScreen.kt:194]: ... copied without thinking too much? As I said, there's already two differently structured copies out there on how the `GameSaver.saveGame` call is wrapped, one for the save screen, one for autosaves (not counting multiplayer) if I remember correctly, and once that code is unified we'd simply call it. Question is - who has the will to attack this monster task...

File: core/src/com/unciv/ui/worldscreen/WorldScreen.kt (hunk 2 above)
Comment (SomeTroglodyte) [reply core/src/com/unciv/ui/worldscreen/WorldScreen.kt:193]: I actually replaced the one with existing translations that would have been adequate - for consistency...

File: core/src/com/unciv/ui/worldscreen/WorldScreen.kt (hunk 3 above)
Comment (SomeTroglodyte) [reply core/src/com/unciv/ui/worldscreen/WorldScreen.kt:199]: Kludge - thread() returns a Thread instance so this casts the return type of the anonymous function to void - ahem, Unit as the dispatcher expects ()->Unit not ()->Thread. There's many alternate solutions, this is maybe the shortest that incurs no additional overhead - but probably you're right it's not obvious. Should the code stay at all I'd simply comment it. An explicit `as` cast would be unchecked...

File: core/src/com/unciv/ui/worldscreen/WorldScreenTopBar.kt (hunk 4 above)
Comment (SomeTroglodyte) [reply core/src/com/unciv/ui/worldscreen/WorldScreenTopBar.kt:152]: Didn't want it to go through the translation engine... But needs comparison with unit commands code anyway (different color at least over there), and possibly meaningless depending on a consensus about my "similar hints for other worldscreen buttons would look horrible" comment. Honestly, I'd _not_ change the button for this.

File: core/src/com/unciv/ui/worldscreen/WorldScreen.kt (hunk 2 above)
Comment (SomeTroglodyte) [reply core/src/com/unciv/ui/worldscreen/WorldScreen.kt:193]: "failed to save game" and "Game saved" do not exist, only "Could not save game!" -> perhaps just tone down the verbosity and omit what does not already have a translation?

_no_. This _needs_ feedback while existing save/load code is self-evident. Well then, equalize fail messages and today I dislike the "OK" - done.
